src/System.Net.NameResolution/tests/PalTests/NameResolutionPalTests.cs

```diff
@@ -166,5 +166,15 @@ public void TryGetNameInfo_LocalHost_IPv6_TryGetAddrInfo()
             Assert.Equal(SocketError.Success, error);
             Assert.NotNull(hostEntry);
         }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        public void Exception_HostNotFound_Success()
+        {
+            var ex = new  SocketException((int)SocketError.HostNotFound);
+
+            Assert.Equal(-1, ex.Message.IndexOf("Device"));
+            Assert.NotEqual(-1,  ex.Message.IndexOf("not known"));
```

Comment:
This won't work on non-English OSes, will it?  And is the terminology guaranteed to be the same across Windows vs macOS vs Linux?  Seems dangerous to depend on the exact terms.

Reply:
We should never be checking for strings like this at all.  For example, these tests won't work in UWP NetNative because string resources are stripped.

Reply:
This is not applicable to Windows/UWP as it is PAL AnyUnix. 

For OSX I did change system language, rebooted and I get same message. (e.g. english)
For Linux I played with LANG and LC_ALL but this always seems to come back in english.

The wording is little bit different between Linux and OSX and that is reason why the test looks for fragments rather than exact match. 

I added test since @stephentoub asked for it. I know this has some potential issues but it works on Linux & OSX as far as I can tell. 

Alternatively I can:
-1  remove the test
-2  check LANG LC_??? and skip test if not C or en_
-3 write C helper function and compare text.

If you have any other specific suggestion please let me know.

Reply:
ok. How about keeping first part to verify it does not  contain word "Device" - like it used to.
And remove second part looking for specific string.

Reply:
Even that's a bit brittle. But it's up to you.